docs/site/Deploying_to_ibm_cloud_kubernetes.md

```diff
@@ -0,0 +1,707 @@
+
+It is important for the host to be ````'0.0.0.0'```` and the port to be consistent in several places. We will use a port of ````8080````.
+
+
+````
+import {DeployTestApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+export {DeployTestApplication};
+
+export async function main(options: ApplicationConfig = {}) {
+
+  // --------- ADD THIS SNIPPET -------------
+if (!options) options = {};
+if (!options.rest) options.rest = {};
+options.rest.port = 8080;
+options.rest.host = "0.0.0.0";
+// ----------------------------------------
+
+const app = new DeployTestApplication(options);
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  console.log(`Try ${url}/ping`);
+
+  return app;
+}
+````
+
+## Step 4: Add .dockerignore file
+
+At project root, create a file called ````.dockerignore```` with the following content:
+
+````
+node_modules
+npm-debug.log
+````
+
+## Step 5: Update package.json
+
+Before we deploy to the cloud, go to package.json at the root of the project directory, and remove the following line:
+
+````
+"prestart": "npm run build",
+````
+
+## Step 6: Build the application
+
+In a command window, navigate the main directory of your project
+and type :
+
+````
+npm run build
+````
+
+Make sure there are no errors before continuing to the next step.
+
+## Step 7: Run the application locally
+
+
+In a command window in the main directory of your project, type:
+
+````
+npm start
+````
+
+The server should start up successfully and display
+
+````
+Server is running at http://127.0.0.1:8080
+Try http://127.0.0.1:8080/ping
+````
+
+Open your browser and attempt to access all these urls
+
+http://127.0.0.1:8080/
+
+http://127.0.0.1:8080/ping
+
+http://127.0.0.1:8080/explorer
+
+http://127.0.0.1:8080/openapi.json
+
+
+![](./imgs/lb4_k8s_ibm_cloud_app_website_1.png)
+
+
+Make sure that the application runs well before continuing to the next step.
+
+In the command window, stop the application with
+
+````
+Ctrl + C
+````
+
+
+## Step 7: Create a Dockerfile
+
+Let's get the application running in Docker.
+
+In the main project directory, create a file named ````Dockerfile```` .
+
+Place this inside:
+
+````
+FROM node:10.14-alpine
+
+EXPOSE 8080
+
+COPY package.json package.json
+RUN npm install
+
+COPY . .
+
+
+RUN npm run build
+
+CMD ["npm", "start" ]
+````
+
+Notice the port is ````8080```` . It matches the port we specified in the application.
+
+
+
+## Step 8: Build a Docker image
+
+This will build a docker image and give it a tag name.
+
+````
+docker build -t lb4-simple-web-app .
+````
+
+When it completes, you will see :
+
+````
+Successfully built 7d26df6c1561
+Successfully tagged lb4-simple-web-app:latest
+````
+
+You can find your image by typing:
+
+```
+docker images | grep lb4-simple-web-app
+```
+
+It will display something like this :
+```
+lb4-simple-web-app   latest 7d26df6c1561
+```
+
```

Comment:
The docker related files can be added as part of `lb4 app` if we add such feature, for example, `lb4 app my-app --docker`. Consider to submit another PR for that.